### PR TITLE
feat: transfer redesign for mobile

### LINKED ISSUE #6460: Transfer Redesign handover
The final design can be found [here](https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=890-14850&t=Ft1lbluGKjwxPDhR-4). 

Feel free to grab it.

image

Ref
---
- https://github.com/kodadot/nft-gallery/issues/6210

_Originally posted by @EthVlad in https://github.com/kodadot/nft-gallery/issues/5575#issuecomment-1600667031_

### LINKED ISSUE #6468: Transfer Redesign handover - (4) Mobile Redesign
related to https://github.com/kodadot/nft-gallery/issues/6460

Task:

 - [ ] mobile redesign
 (feel free to split it into some smaller issues)
image

figma: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?node-id=890%3A14850&mode=dev

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature
- [x] Refactoring

## Needs Design check

- @exezbcz please review
- @kodadot/qa-guild please review

## Context

- [x] Closes #6468 
- [x] Closes #6460 
- [x] Remove old transfer page and `?redesign=true`

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed UI

image

image

### <samp>🤖 Generated by Copilot at 0bdf37f</samp>

Improved the responsiveness and design of the transfer feature for NFTs. Refactored the Transfer and TransferConfirmModal components and added flex layout and mobile detection. Enhanced the NeoModal component to accept multiple custom classes. Removed the old transfer design from the TransferPage component.

### <samp>🤖 Generated by Copilot at 0bdf37f</samp>

> _We are the masters of the NFT_
> _We transfer them with style and speed_
> _We adapt to any device or screen_
> _We are the kings of the Transfer component_

### CODE REVIEW

**Comment by prury** [comment]
looks nice! let's go

**i believe this button doesn't have this square around it**
image

**4 recipients but showing only 2** (they show if you scroll down, but there's a blank space over them)
image

will leave design details for Exez

**Comment by yangwao** [comment]
Koda has own song 🎶 now

**Comment by exezbcz** [comment]
> **i believe this button doesn't have this square around it**

We can either remove completely or just remove the drop shadow - general rule - it is in container that has drop shadow -> no drop shadow on elements inside

**Comment by Jarsen136** [comment]
> 4 recipients but showing only 2 (they show if you scroll down, but there's a blank/dark space over them)

✅ Fixed

>> i believe this button doesn't have this square around it
> We can either remove completely or just remove the drop shadow - general rule - it is in container that has drop shadow -> no drop shadow on elements inside

✅ Done

File: libs/ui/src/components/NeoModal/NeoModal.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
     :destroy-on-hide="destroyOnHide"
     :can-cancel="canCancel"
     :full-screen="fullScreen"
-    :content-class="[contentClass, noShadow ? 'no-shadow' : '']"
+    :content-class="[...contentClassName, noShadow ? 'no-shadow' : '']"
```
Comment (roiLeo) [inline libs/ui/src/components/NeoModal/NeoModal.vue:9]:
```suggestion
    :content-class="[...contentClassName, { 'no-shadow': noShadow }]"
```

**Review by roiLeo — APPROVED** [review]
You really like `<span />` hehe

File: libs/ui/src/components/NeoModal/NeoModal.vue (hunk 1 above)
Comment (Jarsen136) [reply libs/ui/src/components/NeoModal/NeoModal.vue:9]: I tested it but `o-modal` component does not accept this way.

**Comment by yangwao** [comment]
so I'm dispatching payrolls with new UI ? 👀

**Review by preschian — APPROVED** [review]

**Comment by exezbcz** [comment]
Hellooooo
image
remove stroke on sides

- modal background full height, even covering the navbar

- is there an available fund check? I should not be able to send this transaction 
image


otherwise good! thanks

**Comment by Jarsen136** [comment]
> hmm, usd missing?
> 
> image

✅ Fixed. The bug comes from some recent merging : )

**Comment by yangwao** [comment]
> How about a higher mobile modal?

for me yes
cc @exezbcz

**Comment by exezbcz** [comment]
@Jarsen136 I think that does not solve the issue; I would not make it that high, maybe a bit smaller, and it will work - it is the default modal height or? 

thanks! 


cc @EthVlad

**Comment by Jarsen136** [comment]
> @Jarsen136 I think that does not solve the issue; I would not make it that high, maybe a bit smaller, and it will work - it is the default modal height or?
> 
> thanks!
> 
The default max height of NeoModal component is 80% of the viewport. Actually, we do not have a common modal style for mobile device. 


image


Current implementation
image

image

**Comment by exezbcz** [comment]
@Jarsen136 okay good, we can always change it, thanks!

**Comment by yangwao** [comment]
pay 100 usd

let's go!

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 4.52 USD/DOT ~ 22.124 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x382b67cebc16c43673460a925c5d86cb5dc5a451f9a3b9f10a542d9a6b8a5baa](https://polkadot.subscan.io/extrinsic/0x382b67cebc16c43673460a925c5d86cb5dc5a451f9a3b9f10a542d9a6b8a5baa)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by EthVlad** [comment]
>I cannot manage to make some elements cover navbar for the current layout. How about a higher mobile modal?

What do you mean by higher mobile modal? I think that covering 80% is good enough if we could darken the navbar same way as the background. Like here in design, so we focus users attention to modal only. 
Screenshot 2023-08-21 at 10 38 44

**Pull request merged**